tendermint/src/validator.rs

```diff
@@ -30,14 +35,37 @@ impl Set {
             .into_iter()
             .map(|x| x.hash_bytes())
             .collect();
-        let validator_byteslices: Vec<&[u8]> = (&validator_bytes)
-            .into_iter()
-            .map(|x| x.as_slice())
-            .collect();
+        let validator_byteslices: Vec<&[u8]> =
+            (&validator_bytes).iter().map(|x| x.as_slice()).collect();
         merkle::simple_hash_from_byte_slices(validator_byteslices.as_slice())
     }
 }
 
+impl lite::ValidatorSet for Set {
+    type Validator = Info;
+
+    fn hash(&self) -> Hash {
+        // TODO almost the same as above's pub fn hash(self) -> merkle::Hash
+        let validator_bytes: &Vec<Vec<u8>> =
+            &self.validators.iter().map(|x| x.hash_bytes()).collect();
+        let validator_byteslices: Vec<&[u8]> =
+            (&validator_bytes).iter().map(|x| x.as_slice()).collect();
+        Hash::Sha256(merkle::simple_hash_from_byte_slices(
+            validator_byteslices.as_slice(),
+        ))
+    }
+
+    fn total_power(&self) -> u64 {
+        self.validators.iter().fold(0u64, |total, val_info| {
+            total + val_info.voting_power.value()
+        })
+    }
+
+    fn into_vec(&self) -> Vec<Self::Validator> {
```

Comment:
Tried changing this to `iter` in the original traits. It compiles there, but not sure how to implement it yet. See https://github.com/interchainio/tendermint-rs/pull/31#discussion_r330849769.

Maybe should just revert the change since it might not be worth it for now, unless you have an easy way to deal with this.

Reply:
reverted